### PR TITLE
HTCONDOR-2317 default container image

### PR DESCRIPTION
**Pull request by GregThain**
<Insert PR description here, and leave checklist below for code review.>

# HTCondor Pull Request Checklist for internal reviewers

- [ ] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [ ] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [ ] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [ ] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [ ] Verify that the branch destination of the PR matches the target version of the ticket
- [ ] Check for correctness of change
- [ ] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [ ] Check for documentation, if needed  (documentation [build logs](https://readthedocs.org/projects/htcondor/builds/))
- [ ] Check for version history, if needed
- [ ] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [ ] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

File: docs/admin-manual/configuration-macros.rst [hunk 1]
```diff
@@ -6534,6 +6534,17 @@ These settings affect the *condor_starter*.
     to the Singularity command line. This can be an expression evaluted in the context of the
     slot ad and the job ad, where the slot ad is the "MY.", and the job ad is the "TARGET.".
 
+:macro-def:`USE_DEFAULT_CONTAINER[STARTER]`
+    A boolean value or classad expression evaluting to boolean in the context of the Slot
+    ad (the MY.) and the job ad (the TARGET.).  When try, a vanilla universe job that
+    does not request a container will be put into the container image specified By
+    the parameter :macro:`DEFAULT_CONTAINER_IMAGE`
+
+:macro-def:`USE_DEFAULT_CONTAINER[STARTER]`
```
Comment (olifre) [inline docs/admin-manual/configuration-macros.rst:6543]: That should be `DEFAULT_CONTAINER_IMAGE` ;-).

Comment (GregThain) [reply docs/admin-manual/configuration-macros.rst:6543]: Ooops -- thanks for the heads-up.  Will get it fixed right away.

File: docs/admin-manual/configuration-macros.rst [hunk 2]
```diff
@@ -6534,6 +6534,17 @@ These settings affect the *condor_starter*.
     to the Singularity command line. This can be an expression evaluted in the context of the
     slot ad and the job ad, where the slot ad is the "MY.", and the job ad is the "TARGET.".
 
+:macro-def:`USE_DEFAULT_CONTAINER[STARTER]`
+    A boolean value or classad expression evaluting to boolean in the context of the Slot
+    ad (the MY.) and the job ad (the TARGET.).  When try, a vanilla universe job that
+    does not request a container will be put into the container image specified By
```
Comment (JaimeFrey) [inline docs/admin-manual/configuration-macros.rst:6540]:
```suggestion
    A boolean value or classad expression evaluating to boolean in the context of the Slot
    ad (the MY.) and the job ad (the TARGET.).  When true, a vanilla universe job that
    does not request a container will be put into the container image specified by
```

File: docs/version-history/feature-versions-23-x.rst [hunk 3]
```diff
@@ -17,6 +17,11 @@ Release Notes:
 
 New Features:
 
+- Allow the startd to force a job that doesn't ask to run inside a
+  docker or apptainer container inside one with new parameters
+  :param:`USE_DEFAULT_CONTAINER` and :param:`DEFAULT_CONTAINER_IMAGE`
```
Comment (JaimeFrey) [inline docs/version-history/feature-versions-23-x.rst:22]:
```suggestion
  :macro:`USE_DEFAULT_CONTAINER` and :macro:`DEFAULT_CONTAINER_IMAGE`
```

**Review by JaimeFrey — CHANGES_REQUESTED** [review]
Looks good, aside from doc typos.

**Review by JaimeFrey — APPROVED** [review]
LGTM

**Pull request merged**